jwst/refpix/reference_pixels.py

```diff
@@ -181,6 +184,56 @@ def sigma_clip(self, data, dq, low=3.0, high=3.0):
             mean = data[goodpixels].mean(dtype=np.float64)
         return mean
 
+    def transfer_to_model(self, input_model):
+        """
+        Transfer the reference-pixel corrected data arrays from the
+        dataset to the input model
+
+        Parameters:
+        -----------
+
+        input_model: JWST datamodel
+
+        The datamodel the data arrays are to be transferred to
+
+        Returns:
+        --------
+
+        None
+
+        """
+
+        if self.is_subarray:
+            original_size = self.extract_from_embedded()
+            input_model.data = original_size
+            return
+        else:
+            input_model.data = self.data.copy()
+            return
+
+    def extract_from_embedded(self):
+        """
+        Extract the embedded subarray from the full-frame dataset
+        
+        Parameters:
+        -----------
+        
+        None
+        
+        Returns:
+        --------
+        
+        embedded_subarray: nddata array
+        
+        The subarray embedded in the full-frame array
+        
+        """
+        rowstart = self.ystart - 1
+        rowstop = rowstart + self.ysize
+        colstart = self.xstart - 1
+        colstop = colstart + self.xsize
+        embedded_subarray = self.data[:, :, rowstart:rowstop, colstart:colstop]
```

Comment:
Is it ever possible to have the calculation of these slice indexes end up being outside the bounds of the actual array and hence we should implement some kind of checking to make sure that doesn't happen?

Reply:
It's possible, but that would mean that data processing would have had to have introduced the error.  I'll put the check in, it's only a few extra lines.